### PR TITLE
[INLONG-7708][Sort] Fix parse source failed when adding a table during all database migration in Oracle

### LINKED ISSUE #7708: [Bug][Sort] Fix the error that occurs when adding a table during whole database synchronization in Oracle.
### What happened

```java
2023-03-27 16:22:06.074 [debezium-reader-0] INFO  org.apache.inlong.sort.cdc.oracle.shaded.io.debezium.connector.oracle.logminer.LogMinerQueryResultProcessor [] - Table XE.FLINKUSER.VOTE_RECORD_03 is new and will be captured.
2023-03-27 16:22:06.136 [debezium-reader-0] INFO  org.apache.inlong.sort.cdc.oracle.shaded.io.debezium.connector.oracle.OracleConnection [] - Database Version: Oracle Database 11g Express Edition Release 11.2.0.2.0 - 64bit Production
2023-03-27 16:22:07.254 [pool-19-thread-1] INFO  org.apache.inlong.sort.cdc.oracle.shaded.io.debezium.jdbc.JdbcConnection [] - Connection gracefully closed
2023-03-27 16:22:12.404 [debezium-reader-0] ERROR org.apache.inlong.sort.cdc.oracle.shaded.io.debezium.connector.oracle.logminer.LogMinerHelper [] - Mining session stopped due to the org.apache.inlong.sort.cdc.oracle.shaded.org.apache.kafka.connect.errors.DataException: file is not a valid field name
2023-03-27 16:22:12.405 [debezium-reader-0] ERROR org.apache.inlong.sort.cdc.oracle.shaded.io.debezium.pipeline.ErrorHandler [] - Producer failure
org.apache.inlong.sort.cdc.oracle.shaded.org.apache.kafka.connect.errors.DataException: file is not a valid field name
	at org.apache.inlong.sort.cdc.oracle.shaded.org.apache.kafka.connect.data.Struct.lookupField(Struct.java:254) ~[pool-1-thread-10-1679904897888-test-0308sort-connector-oracle-cdc-1.5.0-SNAPSHOT.jar:2.3.0]
	at org.apache.inlong.sort.cdc.oracle.shaded.org.apache.kafka.connect.data.Struct.getCheckType(Struct.java:261) ~[pool-1-thread-10-1679904897888-test-0308sort-connector-oracle-cdc-1.5.0-SNAPSHOT.jar:2.3.0]
	at org.apache.inlong.sort.cdc.oracle.shaded.org.apache.kafka.connect.data.Struct.getString(Struct.java:158) ~[pool-1-thread-10-1679904897888-test-0308sort-connector-oracle-cdc-1.5.0-SNAPSHOT.jar:2.3.0]
	at org.apache.inlong.sort.cdc.oracle.shaded.org.apache.inlong.sort.cdc.base.relational.JdbcSourceEventDispatcher$SchemaChangeEventReceiver.schemaChangeRecordValue(JdbcSourceEventDispatcher.java:190) ~[pool-1-thread-10-1679904897888-test-0308sort-connector-oracle-cdc-1.5.0-SNAPSHOT.jar:2.3.0]
	at org.apache.inlong.sort.cdc.oracle.shaded.org.apache.inlong.sort.cdc.base.relational.JdbcSourceEventDispatcher$SchemaChangeEventReceiver.schemaChangeEvent(JdbcSourceEventDispatcher.java:220) ~[pool-1-thread-10-1679904897888-test-0308sort-connector-oracle-cdc-1.5.0-SNAPSHOT.jar:2.3.0]
	at org.apache.inlong.sort.cdc.oracle.shaded.io.debezium.connector.oracle.OracleSchemaChangeEventEmitter.emitSchemaChangeEvent(OracleSchemaChangeEventEmitter.java:113) ~[pool-1-thread-10-1679904897888-test-0308sort-connector-oracle-cdc-1.5.0-SNAPSHOT.jar:2.3.0]
	at org.apache.inlong.sort.cdc.oracle.shaded.org.apache.inlong.sort.cdc.base.relational.JdbcSourceEventDispatcher.dispatchSchemaChangeEvent(JdbcSourceEventDispatcher.java:149) ~[pool-1-thread-10-1679904897888-test-0308sort-connector-oracle-cdc-1.5.0-SNAPSHOT.jar:2.3.0]
	at org.apache.inlong.sort.cdc.oracle.shaded.org.apache.inlong.sort.cdc.base.relational.JdbcSourceEventDispatcher.dispatchSchemaChangeEvent(JdbcSourceEventDispatcher.java:64) ~[pool-1-thread-10-1679904897888-test-0308sort-connector-oracle-cdc-1.5.0-SNAPSHOT.jar:2.3.0]
	at org.apache.inlong.sort.cdc.oracle.shaded.io.debezium.connector.oracle.logminer.LogMinerQueryResultProcessor.dispatchSchemaChangeEventAndGetTableForNewCapturedTable(LogMinerQueryResultProcessor.java:336) ~[pool-1-thread-10-1679904897888-test-0308sort-connector-oracle-cdc-1.5.0-SNAPSHOT.jar:2.3.0]
	at org.apache.inlong.sort.cdc.oracle.shaded.io.debezium.connector.oracle.logminer.LogMinerQueryResultProcessor.getTableForDmlEvent(LogMinerQueryResultProcessor.java:323) ~[pool-1-thread-10-1679904897888-test-0308sort-connector-oracle-cdc-1.5.0-SNAPSHOT.jar:2.3.0]
	at org.apache.inlong.sort.cdc.oracle.shaded.io.debezium.connector.oracle.logminer.LogMinerQueryResultProcessor.processResult(LogMinerQueryResultProcessor.java:257) ~[pool-1-thread-10-1679904897888-test-0308sort-connector-oracle-cdc-1.5.0-SNAPSHOT.jar:2.3.0]
	at org.apache.inlong.sort.cdc.oracle.shaded.io.debezium.connector.oracle.logminer.LogMinerStreamingChangeEventSource.execute(LogMinerStreamingChangeEventSource.java:280) ~[pool-1-thread-10-1679904897888-test-0308sort-connector-oracle-cdc-1.5.0-SNAPSHOT.jar:2.3.0]
	at org.apache.inlong.sort.cdc.oracle.source.reader.fetch.OracleStreamFetchTask$RedoLogSplitReadTask.execute(OracleStreamFetchTask.java:125) ~[pool-1-thread-10-1679904897888-test-0308sort-connector-oracle-cdc-1.5.0-SNAPSHOT.jar:2.3.0]
	at org.apache.inlong.sort.cdc.oracle.source.reader.fetch.OracleStreamFetchTask.execute(OracleStreamFetchTask.java:73) ~[pool-1-thread-10-1679904897888-test-0308sort-connector-oracle-cdc-1.5.0-SNAPSHOT.jar:2.3.0]
	at org.apache.inlong.sort.cdc.oracle.shaded.org.apache.inlong.sort.cdc.base.source.reader.external.IncrementalSourceStreamFetcher.lambda$submitTask$0(IncrementalSourceStreamFetcher.java:86) ~[pool-1-thread-10-1679904897888-test-0308sort-connector-oracle-cdc-1.5.0-SNAPSHOT.jar:2.3.0]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_332]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_332]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_332]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_332]
	at java.lang.Thread.run(Thread.java:750) [?:1.8.0_332]
```

### What you expected to happen

Successfully synchronized data.

### How to reproduce

Run whole database synchronization in Oracle.

### Environment

_No response_

### InLong version

master

### InLong Component

InLong Sort

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by e-mhui**
### Prepare a Pull Request

[INLONG-7708][Sort] Fix parse source failed when adding a table during all database migration in Oracle

- Fixes #7708

### Motivation

Fix parse source failed when adding a table during all database migration in Oracle.
When adding a new table, the `schemaChangeRecordValue` method reads three values from the `source`: `BINLOG_FILENAME_OFFSET_KEY`, `BINLOG_POSITION_OFFSET_KEY` and `SERVER_ID_KEY`. However, these values only exist in MySQL, so we need to remove them

### Modifications

1. Remove `BINLOG_FILENAME_OFFSET_KEY`, `BINLOG_POSITION_OFFSET_KEY` and `SERVER_ID_KEY` in Oracle.
2. Add judgment for Oracle source record. 

```java
    public static final String MYSQL_SCHEMA_CHANGE_EVENT_KEY_NAME = "io.debezium.connector.mysql.SchemaChangeKey";
    public static final String ORACLE_SCHEMA_CHANGE_EVENT_KEY_NAME = "io.debezium.connector.oracle.SchemaChangeKey";

    /**
     * Whether the source Record is a schema change event.
     * @param sourceRecord
     * @return ture or false
     */
    public static boolean isSchemaChangeEvent(SourceRecord sourceRecord) {
        Schema keySchema = sourceRecord.keySchema();
        return keySchema != null && (MYSQL_SCHEMA_CHANGE_EVENT_KEY_NAME.equalsIgnoreCase(keySchema.name())
                || ORACLE_SCHEMA_CHANGE_EVENT_KEY_NAME.equalsIgnoreCase(keySchema.name()));
    }
```

### CODE REVIEW

**Review by yunqingmoswu — APPROVED** [review]

**Comment by gong** [comment]
@e-mhui `JdbcSourceEventDispatcher` package path is same with cdc-base of oracle-cdc. I think this is not good. It will cause class confict when class load mutil cdc jar.

**Comment by e-mhui** [comment]
> @e-mhui `JdbcSourceEventDispatcher` package path of oracle-cdc is same with cdc-base . I think this is not good. It will cause class conflict when class load mutil cdc jar.

1. Inlong has multiple versions of dependencies for debezium. cdc-base depends on version 1.5.4-final, but Oracle CDC depends on version 1.6.4-final. The method   `HistorizedSchema.storeOnlyCapturedTables()`   is only available in version 1.6.4-final. If we change it to use   `historizedSchema.storeOnlyMonitoredTables()`   in version 1.5.4-final, it may cause conflicts . Therefore, the method is overwritten in Oracle. After all CDCs are upgraded to version 2.3, we can use   `JdbcSourceEventDispatcher`   in cdc-base uniformly.

```java
    @Override
    public void dispatchSchemaChangeEvent(
            TableId dataCollectionId, SchemaChangeEventEmitter schemaChangeEventEmitter)
            throws InterruptedException {
        if (dataCollectionId != null && !filter.isIncluded(dataCollectionId)) {
            // TODO `HistorizedSchema.storeOnlyCapturedTables` first appeared in io.debezium.core: 1.6.4-final,
            // which conflicts with io.debezium.core: 15.4-final of cdc-base,
            // so we implement this method separately in the sub-module.
            // if (historizedSchema == null || historizedSchema.storeOnlyCapturedTables()) {
            if (historizedSchema == null || historizedSchema.storeOnlyMonitoredTables()) {
                LOG.trace("Filtering schema change event for {}", dataCollectionId);
                return;
            }
        }
        schemaChangeEventEmitter.emitSchemaChangeEvent(new SchemaChangeEventReceiver());
    }
```

2. If there are conflicts, I suggest resolving them using the shade.

File: inlong-sort/sort-connectors/cdc-base/src/main/java/org/apache/inlong/sort/cdc/base/util/RecordUtils.java [hunk 1]
```diff
@@ -130,4 +135,24 @@ public static LogicalType convertLogicType(Column column, Struct struct) {
         return null;
     }
 
+    /**
+     * Whether the source Record is a schema change event.
+     * @param sourceRecord
+     * @retur ture or false
```
Comment (EMsnap) [inline inlong-sort/sort-connectors/cdc-base/src/main/java/org/apache/inlong/sort/cdc/base/util/RecordUtils.java:141]: typo

Comment (e-mhui) [reply inlong-sort/sort-connectors/cdc-base/src/main/java/org/apache/inlong/sort/cdc/base/util/RecordUtils.java]: done.

**Comment by gong** [comment]
> > @e-mhui `JdbcSourceEventDispatcher` package path of oracle-cdc is same with cdc-base . I think this is not good. It will cause class conflict when class load mutil cdc jar.
> 
> 1. Inlong has multiple versions of dependencies for debezium. cdc-base depends on version 1.5.4-final, but Oracle CDC depends on version 1.6.4-final. The method   `HistorizedSchema.storeOnlyCapturedTables()`   is only available in version 1.6.4-final. If we change it to use   `historizedSchema.storeOnlyMonitoredTables()`   in version 1.5.4-final, it may cause conflicts . Therefore, the method is overwritten in Oracle. After all CDCs are upgraded to version 2.3, we can use   `JdbcSourceEventDispatcher`   in cdc-base uniformly.
> 
> ```java
>     @Override
>     public void dispatchSchemaChangeEvent(
>             TableId dataCollectionId, SchemaChangeEventEmitter schemaChangeEventEmitter)
>             throws InterruptedException {
>         if (dataCollectionId != null && !filter.isIncluded(dataCollectionId)) {
>             // TODO `HistorizedSchema.storeOnlyCapturedTables` first appeared in io.debezium.core: 1.6.4-final,
>             // which conflicts with io.debezium.core: 15.4-final of cdc-base,
>             // so we implement this method separately in the sub-module.
>             // if (historizedSchema == null || historizedSchema.storeOnlyCapturedTables()) {
>             if (historizedSchema == null || historizedSchema.storeOnlyMonitoredTables()) {
>                 LOG.trace("Filtering schema change event for {}", dataCollectionId);
>                 return;
>             }
>         }
>         schemaChangeEventEmitter.emitSchemaChangeEvent(new SchemaChangeEventReceiver());
>     }
> ```
> 
> 2. If there are conflicts, I suggest resolving them using the shade.

@e-mhui Because package path is same. So shade won't be feasible.

**Comment by gong** [comment]
image
@e-mhui I find a old bug, class path is error

**Review by healchow — APPROVED** [review]

**Comment by e-mhui** [comment]
> > > @e-mhui `JdbcSourceEventDispatcher` package path of oracle-cdc is same with cdc-base . I think this is not good. It will cause class conflict when class load mutil cdc jar.
> > 
> > 
> > 
> > 1. Inlong has multiple versions of dependencies for debezium. cdc-base depends on version 1.5.4-final, but Oracle CDC depends on version 1.6.4-final. The method   `HistorizedSchema.storeOnlyCapturedTables()`   is only available in version 1.6.4-final. If we change it to use   `historizedSchema.storeOnlyMonitoredTables()`   in version 1.5.4-final, it may cause conflicts . Therefore, the method is overwritten in Oracle. After all CDCs are upgraded to version 2.3, we can use   `JdbcSourceEventDispatcher`   in cdc-base uniformly.
> > 
> > ```java
> >     @Override
> >     public void dispatchSchemaChangeEvent(
> >             TableId dataCollectionId, SchemaChangeEventEmitter schemaChangeEventEmitter)
> >             throws InterruptedException {
> >         if (dataCollectionId != null && !filter.isIncluded(dataCollectionId)) {
> >             // TODO `HistorizedSchema.storeOnlyCapturedTables` first appeared in io.debezium.core: 1.6.4-final,
> >             // which conflicts with io.debezium.core: 15.4-final of cdc-base,
> >             // so we implement this method separately in the sub-module.
> >             // if (historizedSchema == null || historizedSchema.storeOnlyCapturedTables()) {
> >             if (historizedSchema == null || historizedSchema.storeOnlyMonitoredTables()) {
> >                 LOG.trace("Filtering schema change event for {}", dataCollectionId);
> >                 return;
> >             }
> >         }
> >         schemaChangeEventEmitter.emitSchemaChangeEvent(new SchemaChangeEventReceiver());
> >     }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > 2. If there are conflicts, I suggest resolving them using the shade.
> 
> @e-mhui Because package path is same. So shade won't be feasible. why not to modify directly package path

done.

**Review by gong — APPROVED** [review]

**Review by EMsnap — APPROVED** [review]

**Pull request merged**